### ISSUE TITLE
qgis does not build on nixos-20.09

### ISSUE THREAD

**Issue by bgamari** [NixOS/nixpkgs#98975]
**Describe the bug**
As of today `qgis-unwrapped` does not build on `nixos-20.09`:

```
[4225/5124] Generating ui_DlgExportVector.pyrlayertestis_rastermarkertest.dir/qgis_rastermarkertest_autogen/mocs_compilation.cpp.oop.o
FAILED: python/plugins/db_manager/ui_DlgExportVector.py
cd /build/source/build/python/plugins/db_manager && /build/source/scripts/pyuic-wrapper.sh /nix/store/656znsafc0bk0sh0q0qwwvdr2d5c1j1k-python3.8-PyQt5-5.14.2/bin/pyuic5 /build/source/build/output/lib /build/source/build/output/pyt>
Cannot mix incompatible Qt library (5.14.2) with this library (5.15.0)
[4226/5124] Generating ui_DlgCancelTaskQuery.py
FAILED: python/plugins/db_manager/ui_DlgCancelTaskQuery.py
cd /build/source/build/python/plugins/db_manager && /build/source/scripts/pyuic-wrapper.sh /nix/store/656znsafc0bk0sh0q0qwwvdr2d5c1j1k-python3.8-PyQt5-5.14.2/bin/pyuic5 /build/source/build/output/lib /build/source/build/output/pyt>
Cannot mix incompatible Qt library (5.14.2) with this library (5.15.0)
[4227/5124] Generating ui_DlgAddGeometryColumn.py
FAILED: python/plugins/db_manager/ui_DlgAddGeometryColumn.py
cd /build/source/build/python/plugins/db_manager && /build/source/scripts/pyuic-wrapper.sh /nix/store/656znsafc0bk0sh0q0qwwvdr2d5c1j1k-python3.8-PyQt5-5.14.2/bin/pyuic5 /build/source/build/output/lib /build/source/build/output/pyt>
Cannot mix incompatible Qt library (5.14.2) with this library (5.15.0)
[4232/5124] Building CXX object tests/src/core/CMakeFiles/qgis_rastermarkertest.dir/testqgsrastermarker.cpp.oop.o
ninja: build stopped: subcommand failed.
builder for '/nix/store/72lasz3gap753vmbx1fbkxph5dhgcj2f-qgis-unwrapped-3.10.9.drv' failed with exit code 1
```

The problem appears to be that `qgis-unwrapped` depends upon both `/nix/store/rz0m4c0ld2vj3cs5afmxwkb78hn0i1jf-python3.8-PyQt5-5.14.2.drv` (which, despite the name, depends upon `/nix/store/lrbhr6d2v9pqzkr5bya6wc73z1ndykff-qtbase-5.15.0.drv`) and `/nix/store/izspngfiasc6r2ww84is130h32hhf2gq-qtbase-5.14.2.drv`.

**To Reproduce**
1. `nix build nixpkgs.qgis-unwrapped`
2. Observe failure

**Expected behavior**
It should build.

**Screenshots**

**Additional context**

**Notify maintainers**
@sander @lsix 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.5.17, NixOS, 20.09beta482.b0a587f2efc (Nightingale)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-20.09beta482.b0a587f2efc, nixpkgs-20.09beta482.b0a587f2efc"`
 - channels(ben): `"nixpkgs-20.09beta482.b0a587f2efc, nixpkgs-unstable-21.03pre243895.72b9660dc18, nixos-18.09-18.09.2574.a7e559a5504"`
 - nixpkgs: `/home/ben/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by sander** [comment]
@bgamari you probably mean to notify another Sander.

**Comment by lsix** [comment]
Yes, pyqt's maintainer is @svanderburg

I’ll try to work on it ASAP.

**Comment by lsix** [comment]
As far as I can tell, this was fixed with 54ef32d2c6e70f1084d10628713bc196b4c4c968

Hydra job now succeeds (https://hydra.nixos.org/build/127991139). I’ll close this issue.

Fee free to re-open if the issue arises again.
